Python/sort/Counting_Sort.py

```diff
@@ -0,0 +1,82 @@
+# Implementation of the Counting Sort algorithm in Python.
+
+# Worst complexity: n+r
+# Average complexity: n+r
+# Space complexity: n+r
+# Stable: Yes
+# Class: Non-comparison sort
+# Data structure: Array
+
+
+def counting_Sort(arr):
+    size = len(arr)
+    output = [0] * size
+
+    count = [0] * 10
+
+    # Store the count of each elements in count array
+    for i in range(0, size):
+        count[arr[i]] += 1
+
+    # Store the cummulative count
+    for i in range(1, 10):
+        count[i] += count[i - 1]
+
+    # Find the index of each element of the original array in count array
+    # place the elements in output array
+    i = size - 1
+    while i >= 0:
+        output[count[arr[i]] - 1] = arr[i]
+        count[arr[i]] -= 1
+        i -= 1
+
+    # Copy the sorted elements into original array
+    for i in range(0, size):
+        arr[i] = output[i]
+
+# Taking input
+n = int(input("Enter length of array (n): "))
+max_val = int(input("Enter the max value in Array (Has to be smaller than n): "))
```

Comment:
You don't need to ask users for the number of elements or maximum elements, as all these can be easily calculated from the list of elements given by the user. Please make the necessary changes and update the sample input/output accordingly.

Reply:
Yes sir I understand I will correct it
Thanks for mentioning